### PR TITLE
CP-18271 first cut at C library for RRD plugins

### PR DESCRIPTION
**Pull request by lindig**
I'd like to invite comments on this first cut to implement a C library that writes the RRD V2 protocol to a file to be consumed by RRDD.  The README (not part of this PR but in the repository) provides a design overview.

The commit history is messy as I used it just for snapshots; it will be squashed later.

### CODE REVIEW

File: librrd.h [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+
+#include <stdint.h>
+#include "parson/parson.h"
+
+#define RRD_MAX_SOURCES         128
+
+#define RRD_OK                  0
+#define RRD_TOO_MANY_SOURCES    1
+#define RRD_NO_SOUCH_SOURCE     2
```
Comment (robhoes) [inline librrd.h]: Souch?

**Comment by Zhengchai** [comment]
Thanks. What is the interface output difference comparing with the current one?

Asking this because of many other components like WLB who are using the RRD output also, keep the same output formats and data will not cause regression.

Also exercise with a full WLB tests and understand the results will be great.

**Comment by lindig** [comment]
I don't understand the question. The file format for communicating with RRDD is specified here: https://info.citrite.net/display/engp/RRD+plugin+protocol+V2 and this is what the library aims to implement.

**Comment by Zhengchai** [comment]
OK, in a simple word, with V2 RRD, is there any change of existing RRD RPC exposing/Providing data structure and parameters? e.g. When Client requesting data from host with command "HTTP://x.x.x.x:80/rrd_updates?session_id=OpaqueRef:xxxxxxxx&cf=AVERAGE&start=1470036782&host=true", will the output be changed?

I can post your an example of current result if you need to comparing (e.g. with email).

**Comment by lindig** [comment]
I don't think this is relevant here. This protocol and library is about data sources reporting data to RRDD and not about how RRDD presents the information it has collected to the outside. So I don't expect any change to current behaviour.

**Comment by Zhengchai** [comment]
That will be great if no impacts/changes the output side. just want to ensure this given that kind of change may impact a lot to WLB indeed. thanks for confirming. Let me know please if that is not the case. thanks.

**Comment by robhoes** [comment]
I think it's great! :)
:shipit:

**Comment by lindig** [comment]
I've started testing it with rrd-transport and there are some bugs that I need to weed out first in the data format.

**Pull request merged**